mypy_django_plugin/django/context.py

```diff
@@ -369,6 +369,9 @@ def get_field_related_model_cls(self, field: Union["RelatedField[Any, Any]", For
         else:
             related_model_cls = field.field.model
 
+        if related_model_cls is None:
+            raise UnregisteredModelError
+
```

Comment:
I would like it if we cleaned up the if statements here if you wouldn't mind changing a couple more lines?

Essentially raise error in 1 place at the end e.g.

```python
        related_model_cls = None
        if isinstance(field, RelatedField):
            related_model_cls = field.remote_field.model
        else:
            related_model_cls = field.field.model

        if isinstance(related_model_cls, str):
            if related_model_cls == "self":  # type: ignore[unreachable]
                # same model
                related_model_cls = field.model
            elif "." not in related_model_cls:
                related_model_fullname = f"{field.model.__module__}.{related_model_cls}"
                related_model_cls = self.get_model_class_by_fullname(related_model_fullname)
            else:
                with contextlib.suppress(LookupError):
                    related_model_cls = self.apps_registry.get_model(related_model_cls)

        if related_model_cls is None:
            raise UnregisteredModelError
        return related_model_cls
```

Reply:
✅

Reply:
Oh, my suggestion wouldn't work actually. Since we write `related_model_cls` to a string.

But it's interesting that we have an unreachable ignore too, perhaps that hides something.

Anyways, unless we keep 2 variables I'm not sure we could change to my suggestion

Reply:
I recently went through this code to figure things out. It's a bit hard to understand what Django is doing, but from what I've experienced `field.remote_field.model` can't be a string unless the string reference can't be resolved (which should fail at runtime).

I'm doing things a bit differently from the mypy plugin, but you can find the diff of how I handle it [here](https://github.com/Viicos/django-autotyping/commit/416814bd084a670e2a116b3869dd203914bff018#diff-58b11942d8703e587bdc6b64e799314b2c70974f3fe6f3e3aea3e679073bd13cR138-R143).

(So the unreachable ignore is probably because the type hint says `field.remote_field.model` is always `type[Model]`, not `str`)

Reply:
Yeah, Django probably sanitizes the field/model attributes. Input is string but never after instantiation.

I suspect the unreachable is easy to get rid of if we remove the type inference, since mypy can't be aware of the above happening.

Something like this:

```python
def get_field_related_model_cls(...) -> Type[Model]:
    related_model_cls: Type[Model] | str | None = None
    ...
```

It doesn't allow for my suggestion above to work or really help out with anything other than getting rid of the unreachable though.